extensions/cdc-mysql/src/test/java/com/hazelcast/jet/cdc/mysql/MySqlCdcNetworkIntegrationTest.java

```diff
@@ -139,31 +151,34 @@ public void when_networkDisconnectDuringSnapshotting_then_jetSourceIsStuckUntilR
                 ToxiproxyContainer toxiproxy = initToxiproxy(network);
         ) {
             MySQLContainer<?> mysql = initMySql(network, null);
-            ToxiproxyContainer.ContainerProxy proxy = initProxy(toxiproxy, mysql);
-            Pipeline pipeline = initPipeline(proxy.getContainerIpAddress(), proxy.getProxyPort());
-            // when job starts
-            JetInstance jet = createJetMembers(2)[0];
-            Job job = jet.newJob(pipeline);
-            assertJobStatusEventually(job, RUNNING);
+            try {
+                ToxiproxyContainer.ContainerProxy proxy = initProxy(toxiproxy, mysql);
+                Pipeline pipeline = initPipeline(proxy.getContainerIpAddress(), proxy.getProxyPort());
+                // when job starts
+                JetInstance jet = createJetMembers(2)[0];
+                Job job = jet.newJob(pipeline);
+                assertJobStatusEventually(job, RUNNING);
 
-            // and snapshotting is ongoing (we have no exact way of identifying
-            // the moment, but random sleep will catch it at least some of the time)
-            TimeUnit.MILLISECONDS.sleep(ThreadLocalRandom.current().nextInt(0, 500));
+                // and snapshotting is ongoing (we have no exact way of identifying
+                // the moment, but random sleep will catch it at least some of the time)
+                TimeUnit.MILLISECONDS.sleep(ThreadLocalRandom.current().nextInt(0, 500));
```

Comment:
Not related to this PR, but could it be made more deterministic by having a `CountDownLatch`, call `await` here and CountDownLatch#countDown in a first step of the pipeline (peek/map)?

Reply:
In this test I'm trying to see what happens if the connection to the DB drops while snapshotting is going on. Snapshotting is a process with 9 stages. I'm interested to see what happens in all the stages, on connection drop. If I would make the test more deterministic (for example the way you suggest), then it would probably always trigger around the same stage, towards the end or after snapshotting. I don't want that. With such a random sleep there will eventually always be runs for all possible scenarios. Yes, it's not deterministic, will be hard to debug, but if there is a problem we will likely find out about it. Eventually. This was my thinking.